contracts/examples/bonding-curve/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+elrond_wasm::imports!();
+elrond_wasm::derive_imports!();
+
+mod bc_function;
+use bc_function::{BCFunction, CurveArguments, SupplyType, Token};
+
+mod linear_function;
+use linear_function::LinearFunction;
+
+const TOKEN_NUM_DECIMALS: usize = 18;
+
+#[elrond_wasm_derive::contract]
+pub trait BondingCurve {
+	#[init]
+	fn init(&self, issued_token: Token, exchanging_token: Token) {
+		self.issued_token().set(&issued_token);
+		self.exchanging_token().set(&exchanging_token);
+	}
+
+	// endpoint - owner-only
+
+	#[payable("*")]
+	#[endpoint(issueToken)]
+	fn issue_token(
+		&self,
+		initial_supply: Self::BigUint,
+		#[payment] issue_cost: Self::BigUint,
+	) -> SCResult<AsyncCall<Self::SendApi>> {
+		only_owner!(self, "only owner may call this function");
+
+		require!(
```

Comment:
This will always fail, as you've already set a value for issued_token in the #init function.